app/lib/dates.lib.js

```diff
@@ -0,0 +1,106 @@
+'use strict'
+
+/**
+ * General helper methods
+ * @module DatesLib
+ */
+
+const february = 2
+const lastDayOfFebruary = 28
+const lastDayOfFebruaryLeapYear = 29
+
+/**
+ * Formats a string assumed to be a date in the format 01/01/2001
+ *
+ * Formats to iso format 2001-01-01
+ *
+ * This will format a date that may not be valid as it is just string manipulation
+ *
+ * @param {string} date - The date in the format 01/01/2001
+ * @returns {string | null} - a date in the iso format 2001-01-01
+ */
+function formatStandardDateToISO (date) {
+  if (date === 'null' || date === null) {
+    return null
+  }
+
+  const [day, month, year] = date.split('/')
+
+  const isoDateString = `${year}-${month}-${day}`
+
+  if (isValidDate(isoDateString)) {
+    return isoDateString
+  } else {
+    return null
+  }
+}
+
+/**
+ * Check if a date is valid or not by creating a date and checking it gives the time
+ *
+ * @param {dateString | undefined } dateString - The date in the iso format 2001-01-01
+ * @returns {boolean}
+ */
+function isValidDate (dateString) {
+  if (!dateString) {
+    return false
+  }
+
+  if (!_isValidLeapYearDate(dateString)) {
+    return false
+  }
+
+  const date = new Date(dateString)
+
+  return !isNaN(date.getTime())
+}
+
+/**
+ * Checks a string matches the ISO 8601 date format
+ *
+ * @param {dateString} dateString - The date in the iso format 2001-01-01
+ * @returns {boolean}
+ */
+function isISODateFormat (dateString) {
```

Comment:
I think this should throw an error.

I think we want this functionally to highlight any date issues. 

And we place it wherever a date string should be in the correct format. (mainly in this dates lib)